src/ui/map.ts

```
@@ -2176,6 +2177,43 @@ export class Map extends Camera {
         return this.style.getSource(id) as TSource;
     }
 
+    /**
+     * Change the tile LOD behavior of the specified source. These parameters have no effect when 
+     * pitch == 0, and the largest effect when the horizon is visible on screen.
+     *
+     * @param maxZoomLevelsOnScreen - The maximum number of distinct zoom levels allowed on screen at a time.
+     * There will generally be fewer zoom levels on the screen, the maximum can only be reached when the horizon
+     * is at the top of the screen. Increasing the maximum number of zoom levels causes the zoom level to decay 
+     * faster toward the horizon.
+     * @param tileCountMaxMinRatio - The ratio of the maximum number of tiles loaded (at high pitch) to the minimum
+     * number of tiles loaded. Increasing this ratio allows more tiles to be loaded at high pitch angles. If the ratio
+     * would otherwise be exceeded, the zoom level is reduced uniformly to keep the number of tiles within the limit.
+     * @param id - The ID of the source to set tile LOD parameters for. All sources will be updated if unspecified.
+     * @returns True is success, false if failure (for example if the ID
```

Should it throw instead of returning false? What is the behavior for other methods?

---

Some methods, throw, others return `undefined` or an empty array. Given that this method does not have a natural return value, I've updated it to throw.